comp/otelcol/otlp/components/statsprocessor/agent.go

```diff
@@ -36,14 +38,26 @@ type TraceAgent struct {
 	exit chan struct{}
 }
 
+type statsAdder interface {
+	Add(*pb.StatsPayload)
+}
+
+type noopTraceWriter struct{}
+
+func (n *noopTraceWriter) Stop() {}
+
+func (n *noopTraceWriter) WriteChunks(_ *writer.SampledChunks) {}
+
+func (n *noopTraceWriter) FlushSync() error { return nil }
+
 // NewAgent creates a new unstarted traceagent using the given context. Call Start to start the traceagent.
 // The out channel will receive outoing stats payloads resulting from spans ingested using the Ingest method.
-func NewAgent(ctx context.Context, out chan *pb.StatsPayload, metricsClient statsd.ClientInterface, timingReporter timing.Reporter) *TraceAgent {
+func NewAgent(ctx context.Context, out statsAdder, metricsClient statsd.ClientInterface, timingReporter timing.Reporter) *TraceAgent {
```

Comment:
Perhaps these interfaces should be public. Otherwise OTel will not be able to look it up in the docs.

Do we want to expose the `noopTraceWriter` as well?